modules/soc_eq/main.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+OPENWBBASEDIR=$(cd `dirname $0`/../../ && pwd)
+RAMDISKDIR="$OPENWBBASEDIR/ramdisk"
+MODULEDIR=$(cd `dirname $0` && pwd)
+CHARGEPOINT=$1
+
+case $CHARGEPOINT in
+  2)
+    # second charge point
+    ladeleistungfile="$RAMDISKDIR/llaktuells1"
+    soctimerfile="$RAMDISKDIR/soctimer1"
+    soc_file="$RAMDISKDIR/soc1"
+    soc_eq_client_id=$soc_eq_client_id_lp2
+    soc_eq_client_secret=$soc_eq_client_secret_lp2
+    soc_eq_vin=$soc_eq_vin_lp2
+    ;;
+  *)
+    # defaults to first charge point for backward compatibility
+    # set CHARGEPOINT in case it is empty (needed for logging)
+    CHARGEPOINT=1
+    ladeleistungfile="$RAMDISKDIR/llaktuell"
+    soctimerfile="$RAMDISKDIR/soctimer"
+    soc_file="$RAMDISKDIR/soc"
+    soc_eq_client_id=$soc_eq_client_id_lp1
+    soc_eq_client_secret=$soc_eq_client_secret_lp1
+    soc_eq_vin=$soc_eq_vin_lp1
+    ;;
+esac
+
+
+soctimer=$(<$soctimerfile)
+ladeleistung=$(<$ladeleistungfile)
+
+if (( ladeleistung > 500 ));then
+  tmpintervall=$(( 5 * 6 ))
+else
+  tmpintervall=$(( 60 * 6 ))
+fi
+
+if (( soctimer < tmpintervall )); then
+  soctimer=$((soctimer+1))
+  echo $soctimer > $soctimerfile
+else
+  export soc_eq_client_id soc_eq_client_secret soc_eq_vin soc_file CHARGEPOINT
+  /var/www/html/openWB/modules/soc_eq/soc.py
```

Comment:
Bitte an den Aufruf der soc.py ein "&" anhängen, damit das Skript im Hintergrund ausgeführt wird und nicht die Regelschleife ausbremst.

Reply:
Gibt es ein SoC Modul, das als gutes Beispiel dienen kann? Ich hab den Code in einem Beitrag im Forum  bekommen und bei mir eingebaut. Das hat soweit funktioniert, ein paar Blicke habe ich schon in andere SoC Module geworfen, aber da sind die Implementierungen auch sehr unterschiedlich.

Reply:
Da hast Du Recht, das ist teilweise sehr unterschiedlich. Schau mal dort rein, das entspricht dem, was mir als "Standard" vorschwebt: https://github.com/snaptec/openWB/blob/6cc2ca384c877b3b3b936995e6b91ad90916b41d/modules/soc_kia/main.sh#L56

Reply:
ist analog realisiert, allerdings verstehe ich and er Stelle den sudo nicht. Ist der zwingend? Bei meinem Modul geht das auch ohne sudo, allerdings als Installation auf einem meiner Raspis.

Reply:
Welchen OS User verwendet denn eine openWB Installation auf dem Raspi? Ich hatte heute ein Thema mit Userberechtigungen auf Files um die Token zu persistieren. Die Callback URL ist bei mir ein php Script, welches dann ein Python Script aufruft. Die laufen auf meinem Raspi wie üblich unter dem User www-data. Die openWB Verzeichnisse liegen bei mir aber unter dem User pi.
VG
Det

Reply:
OpenWB nutzt sudo häufig und auch meiner Meinung nach zu oft. Alle Prozesse laufen als user pi. Mit Version 2.x werden die benötigten Rechte detaillierter geprüft. Wenn Dein Modul ohne sudo läuft, kann es gerne weggelassen werden.

Reply:
Läuft der Apache auch als User pi? Ansonsten müssen die leeren Dateien zum Persistieren der Token und der Expire Time  mit user pi und  mode 666 angelegt sein, damit der apache im callback auch darauf zugreifen darf.

VG
Det